app/controllers/goods_nomenclatures_controller.rb

```diff
@@ -17,7 +17,7 @@ def find_relevant_goods_code_or_fallback
     results = @search.perform
 
     if results.exact_match?
-      redirect_to url_for(results.to_param.merge(url_options))
+      redirect_to url_for(results.to_param.merge(url_options).merge(only_path: true))
```

Comment:
An alternative approach would be to use the `:host` option.
But I am not sure which approach is better.

Reply:
Whats the advantage? We'd need to use `host: request.host` otherwise we'd potentially still run into the same issue (ie `request.host != default_url_options[:host]`)

If we just use internal paths only the problem goes away?